flight/search/cell_map.py

```diff
@@ -0,0 +1,140 @@
+"""
+Summary
+-------
+Defines the CellMap class and has some basic tests at the bottom of the file.
+"""
+
+from cell import Cell
+from helper import get_bounds
+
+
+class CellMap:
+    """
+    a two-dimensional array where the row element's are square segments of the
+    search area, each associated with some probability of having an ODLC
+
+    Attributes
+    ----------
+    data : list[list[Cell]
+        a two-dimensional array of Cell objects that defines
+        the search area
+    num_valids : int
+        the number of Cells within the bounds,
+    bounds: dict[str, list[float]]
+        the latitude longitude boundaries of the search area
+
+    Methods
+    -------
+    __get_valids(points: list[list[tuple[float, float] | str]]) -> int
+        returns the number of valid points in the points passed to the CellMap
+    __init_map(points: list[list[tuple[float, float] | str]], odlc_count: int)-> list[list[Cell]]
+        creates two-dimensional list of Cells
+    display()
+        prints state of CellMap into standard output
+    """
+
+    def __get_valids(self, points: list[list[tuple[float, float] | str]]) -> int:
+        """
+        returns the number of valid points in the points passed to the CellMap
+
+        Parameters
+        ----------
+        points : list[list[tuple[float, float] | str]]
+            the points that define the CellMap's data
+
+        Returns
+        -------
+        num_valids : int
+            the number of valid cells in the cell map
+        """
+
+        count: int = 0
+        i: int
+        j: int
+        for i, _ in enumerate(points):
+            for j, _ in enumerate(points[0]):
+                if points[i][j] != "X":
+                    count += 1
```

Comment:
Don't need to `enumerate` if you are discarding the item element (that's what you are doing when you do `i, _`). Use `range` instead.
```suggestion
        for i in range(len(points)):
            for j in range(len(points[0])):
                if points[i][j] != "X":
                    count += 1
```